x-pack/legacy/plugins/maps/public/shared/layers/sources/es_source.js

```diff
@@ -279,4 +281,8 @@ export class AbstractESSource extends AbstractVectorSource {
     return true;
   }
 
+  getId() {
```

Comment:
would it make sense to enforce id creation in a createDescriptor/constructor of the abstractvectorsource (?) it's a little sprinkled throughout right now.

Reply:
I think would be a good idea. I recommend putting that in a separate PR because it might get a little complicated.